src/argilla_sdk/settings/_resource.py

```diff
@@ -176,22 +176,23 @@ def get(self) -> "Settings":
     def create(self) -> "Settings":
         self.validate()
 
-        self._create_fields()
-        self._create_vectors()
-        self._create_metadata()
-        self._create_questions()
         self._update_dataset_related_attributes()
+        self.__fields.create()
+        self._create_questions()
+        self.__vectors.create()
+        self.__metadata.create()
 
         self._update_last_api_call()
         return self
 
     def update(self) -> "Resource":
         self.validate()
 
-        self._upsert_fields()
-        self._upsert_vectors()
-        self._upsert_metadata()
         self._update_dataset_related_attributes()
+        self.__fields.update()
+        self.__vectors.update()
+        self.__metadata.update()
+        # self.questions.update()
```

Comment:
What's the plan with questions?

Reply:
The questions would be refactorized in the same way we use fields, vectors and metadata. But I would like to tackle this as a separate work since questions may have more changes to achieve this.

With this, we can start playing around with the new update dataset settings workflow while we work with question changes.